src/graphics/program-lib/programs/standard-node.js

```diff
@@ -0,0 +1,93 @@
+import { standard } from './standard.js';
+
+// just do a shallow copy standard
+var standardnode = Object.assign({}, standard);
+
+// override a couple of functions - the overridden functions have been altered in standard to accept extra params
+standardnode._generateKey = standard.generateKey;
+standardnode.generateKey = function (options) {
+    return this._generateKey(options, options._shaderGraphChunk.id);
+};
+
+standardnode._createShaderDefinition = standard.createShaderDefinition;
+standardnode.createShaderDefinition = function (device, options) {
+    var rootShaderGraph = null;
+    var rootDeclGLSL = '';
+    var rootCallGLSL = '';
+
+    if (options._shaderGraphChunk) {
+        rootShaderGraph = options._shaderGraphChunk;
+
+        rootDeclGLSL = rootShaderGraph.generateRootDeclGlsl();
+        rootCallGLSL = rootShaderGraph.generateRootCallGlsl();
+    }
+
+    var graphCode = [];
```

Comment:
instead of array which we have fixed 4 slots in that we access as array[1] and similar, make it an object with more descriptive names perhaps, making it easier on receiver side to understand the code. For example:
```
var graphCode = {};
graphCode.definitions = ...;
graphCode.functions = ...;
```

Reply:
makes sense - I will improve this

Reply:
I have improved this - I haven't turned it into a generic data driven system because it is still tied to the non-data driven standard material - and IMO it would become harder to maintain/debug.